cobbler/remote.py

```diff
@@ -3911,6 +3972,16 @@ def get_tftp_file(
         self.check_access(token, "get_tftp_file")
         return self.api.get_tftp_file(path, offset, size)
 
+    def begin_transaction(self, token: str) -> bool:
+        self.transactions[token] = {}
+        return True
+
+    def commit_transaction(self, token: str) -> bool:
+        try:
+            self.api.add_remove_items(list(self.transactions[token].values()))
```

Comment:
A transaction is more than add and remove. Modifying inside a transaction is a completely valid operation. Save calls also need to be caught and switched to no-op as we want to save after the transaction (otherwise you regenerate in the background without changes).

Reply:
The list contains items marked for deletion or the final state of modified items. add_remove_items saves the changes and makes them permanent. 
Save calls are handled.